tensorflow-utils/src/main/java/org/tensorflow/util/buffer/DataBuffer.java

```diff
@@ -0,0 +1,138 @@
+/*
+ Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ =======================================================================
+ */
+
+package org.tensorflow.util.buffer;
+
+import java.nio.ReadOnlyBufferException;
+
+/**
+ * A container of data of a specific type.
+ *
+ * <p>Instances of {@code DataBuffer} map native or heap memory segments to a linear view that
+ * supports:
+ * <ul>
+ *  <li>64-bits indexation, allowing to work with a single buffer with a size up to
```

Comment:
It's probably limited to 2^48 as there are only 48 address bits in current x86_64 hardware. The top address bits tend to be used for access control or pointer colouring, and so we can't usually get into them.